salt/modules/saltutil.py

```diff
@@ -1444,6 +1445,14 @@ def _exec(client, tgt, fun, arg, timeout, tgt_type, ret, kwarg, **kwargs):
             'ret': ret, 'cli': True, 'kwarg': kwarg, 'sub': kwargs['subset'],
         }
         del kwargs['subset']
+    elif kwargs.get('asynchronous'):
+        cmd_kwargs = {
```

Comment:
Maybe we should define `cmd_kwargs` above the if statements, then just modify it with the differences in the respective if clauses.

On a related note, are the clauses `batch`, `subset` and `asynchronous` mutually exclusive? @waynew what do you think?